compiler/luci-interpreter/src/kernels/Concatenation.cpp

```diff
@@ -69,11 +69,32 @@ void Concatenation::configure()
   Shape output_shape = t0->shape();
   output_shape.dim(axis) = sum_axis;
 
-  // TODO S8 type needs more checking: quantization parameters of all input tensors and the output
-  //  tensor should be the same. Note that there is no such requirement for U8 type.
-  if (t0->element_type() == DataType::S8)
-    throw std::runtime_error("Unsupported type.");
-
+  // If input tensors are INT8 type then quantization parameters of all input tensors and the output
+  // should be the same
+  for (int32_t i = 0; i < num_inputs; ++i)
+  {
+    const Tensor *current_tensor = _inputs.at(i);
+    if (current_tensor->element_type() == DataType::S8)
+    {
+      LUCI_INTERPRETER_CHECK(current_tensor->quantized_dimension() ==
+                             output()->quantized_dimension());
+      const auto quantized_dimension = current_tensor->quantized_dimension();
+
+      LUCI_INTERPRETER_CHECK(current_tensor->zero_points().at(quantized_dimension) ==
+                             output()->zero_points().at(quantized_dimension));
+      LUCI_INTERPRETER_CHECK(current_tensor->scales().at(quantized_dimension) ==
+                             output()->scales().at(quantized_dimension));
+      for (int32_t j = i + 1; j < num_inputs; ++j)
+      {
+        const Tensor *second_tensor = _inputs.at(j);
+        LUCI_INTERPRETER_CHECK(quantized_dimension == second_tensor->quantized_dimension());
+        LUCI_INTERPRETER_CHECK(current_tensor->zero_points().at(quantized_dimension) ==
+                               second_tensor->zero_points().at(quantized_dimension));
+        LUCI_INTERPRETER_CHECK(current_tensor->scales().at(quantized_dimension) ==
+                               second_tensor->scales().at(quantized_dimension));
+      }
```

Comment:
You are already compared all input tensors with output, this loop is not needed.